sound/soc/sof/sof-client.c

```diff
@@ -305,7 +305,8 @@ EXPORT_SYMBOL_NS_GPL(sof_client_ipc_tx_message, SND_SOC_SOF_CLIENT);
 
 int sof_client_ipc_rx_message(struct sof_client_dev *cdev, void *ipc_msg, void *msg_buf)
 {
-	if (cdev->sdev->pdata->ipc_type == SOF_IPC_TYPE_3) {
+	if (IS_ENABLED(CONFIG_SND_SOC_SOF_IPC3) &&
```

Comment:
@plbossart does it make sense to allow the IPC3 type when the config is not selected in the first place?

Reply:
I am not sure I understand what you mean by "allow the IPC3 type" @ranj063. Maybe you are referring to another layer of errror control at a higher level, but for this PR the only goal is to fix the build. The IPC type is only known at run-time, different problem IMHO.

Reply:
Fixes: 70dad53ddff0 ("ASoC: SOF: Add IPC3 Kernel Injector")

I would use `#ifdef CONFIG_SND_SOC_SOF_IPC3` around this instead.

Reply:
Hrm, simple backport will not  work due to the IPC generic naming change, but I would do this for the rebase branch and re-arrange t he patches and fix them up

Reply:
Takashi and Mark have suggested multiple times to use IS_ENABLED since it makes the code look 'better', I only use the macros when the compiler is too stupid.

Reply:
> Hrm, simple backport will not work due to the IPC generic naming change, but I would do this for the rebase branch and re-arrange t he patches and fix them up

we can do this after merging and making sure that topic/sof-dev and topic/sof-dev-rebase are the same after the patch reshuffling.

Reply:
> Takashi and Mark have suggested multiple times to use IS_ENABLED since it makes the code look 'better', I only use the macros when the compiler is too stupid.

IS_ENABLED is good in case when a config option is three state, for bool it does not really matter.

But my question was more about the
```c
	if (IS_ENABLED(CONFIG_SND_SOC_SOF_IPC3) &&
	    cdev->sdev->pdata->ipc_type == SOF_IPC_TYPE_3) {
		struct sof_ipc_cmd_hdr *hdr = ipc_msg;

		if (hdr->size < sizeof(hdr)) {
			dev_err(cdev->sdev->dev, "The received message size is invalid\n");
			return -EINVAL;
		}

		sof_ipc3_do_rx_work(cdev->sdev, ipc_msg, msg_buf);
		return 0;
	}
```
vs
```c
#if IS_ENABLED(CONFIG_SND_SOC_SOF_IPC3)
	if (cdev->sdev->pdata->ipc_type == SOF_IPC_TYPE_3) {
		struct sof_ipc_cmd_hdr *hdr = ipc_msg;

		if (hdr->size < sizeof(hdr)) {
			dev_err(cdev->sdev->dev, "The received message size is invalid\n");
			return -EINVAL;
		}

		sof_ipc3_do_rx_work(cdev->sdev, ipc_msg, msg_buf);
		return 0;
	}
#endif
```

Probably the compiler will just drop the whole if statement if IPC3 is not enabled so the linker will not see the `sof_ipc3_do_rx_work()`?
fwiw, this will fail to compile:
```c
	if (0) {
		this will fail to compile
	}
```